projects: add eval-pqmon

rbudai98 opened the pull request:
## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

buha commented on projects/eval-pqmon/Makefile:
```diff
@@ -0,0 +1,22 @@
+STACK_SIZE=0x20000
+
+# Define PQM libray path here:
+PQLIB_PATH=''
```
the build should be invokable with `make PQLIB_PATH=/my/path`  or similar without editing this file

buha commented on projects/eval-pqmon/src/common/afe_config.c:
```diff
@@ -0,0 +1,514 @@
+	int32_t status = 0;
+	uint32_t config;
+
+	// status = timer_init();
```
remove this commented out code if not needed anymore

buha commented on projects/eval-pqmon/src/common/afe_config.c:
```diff
@@ -0,0 +1,514 @@
+
+int32_t afe_init(void)
+{
+	int32_t status = 0;
```
use int instead of int32_t for status (we generally use stdint types but we make an exception for status return values)

buha commented on projects/eval-pqmon/src/common/afe_config.c:
```diff
@@ -0,0 +1,514 @@
+/************************* Functions Definitions ******************************/
+/******************************************************************************/
+
+int32_t afe_init(void)
```
use `int` whenever you return a status

buha commented on projects/eval-pqmon/src/common/afe_config.c:
```diff
@@ -0,0 +1,514 @@
+		}
+	}
+
+	if (status == 0) {
```
i see you're using the alternate method for handling return statuses

there is nothing wrong with it but it just looks at odds with the rest of code in the no-OS repo

buha commented on projects/eval-pqmon/src/common/afe_config.c:
```diff
@@ -0,0 +1,514 @@
+			 BITM_WFB_CFG_BURST_ALL_CHAN;
+		status = afe_write_16bit_reg(REG_WFB_CFG, (uint16_t *)&config);
+		if (status != 0) {
+			status = SYS_STATUS_AFE_WFB_CFG_FAILED;
```
no need for {} brackets here (and in many other places too) but as you prefer

buha commented on projects/eval-pqmon/src/common/afe_config.c:
```diff
@@ -0,0 +1,514 @@
+		frequency = BITM_ACCMODE_SELFREQ_50HZ;
+	} else if (frequency == 60) {
+		frequency = BITM_ACCMODE_SELFREQ_60HZ;
+	}
```
you probably need an else return -EINVAL here in case frequency is not 50 and not 60

buha commented on projects/eval-pqmon/src/common/common_data.c:
```diff
@@ -0,0 +1,184 @@
+	.extra = SPI_PQM_EXTRA,
+}; // SPI init param for ade9430
+
+// struct no_os_timer_init_param timer_ip = {
```
remove unused commented-out code

buha commented on projects/eval-pqmon/src/common/iio_pqm.c:
```diff
@@ -0,0 +1,1039 @@
+ * @return          - The length of the buffer in case of success, negative
+ * value otherwise
+ */
+int write_flicker_model_attr(void *device, char *buf, uint32_t len,
```
you don't need a function for each read and each write of each attribute, you can do this with a single read function and a single write function that manages all attributes and channels at once

buha commented on projects/eval-pqmon/src/common/iio_pqm.c:
```diff
@@ -0,0 +1,1039 @@
+	switch (channel->type) {
+	case IIO_VOLTAGE:
+		switch (attr_id) {
+		case 0: // Voltage RMS value
```
add some #defines or an enum for these magic numbers

rbudai98 commented:
v2:
* applied PR suggestions
* added process_data (for disabling 10-12 cycle processes) and fw_version device attributes
* added THD channel attributes
* changed waveform_buffer to circular buffer (for rolling mode)

buha commented:
CI fails when building pqlib documentation

rbudai98 commented:
v3:
* removed external references for doxygen build
